### PR TITLE
report: specify transfer or resource size for columns

### LINKED ISSUE #10357: unclear that byte / resource sizes are "transfer size"
the following parts of the report display KBs which refer to the _transferSize_ (not _resourceSize_) of a resource:

* performance budgets (resource-summary)
* all of the byte efficiency audits

Not only do we not explicitly say "these values are transfer sizes", we also have copy that is misleading. "File Size" makes me think these would be resource sizes.

image

We should make it clear what these values are (post-compression / wire size / w.e).

____

Instead of just improving the copy, what if we displayed both (or had a toggle)? I find thinking of CSS / JS / anything in terms of transfer size to be unintuitive. For JS specifically, I think resource size might correlate better with performance than transfer size does (IG has an article supporting this https://instagram-engineering.com/making-instagram-com-faster-code-size-and-execution-optimizations-part-4-57668be796a8)

### PR DESCRIPTION
**Pull request by connorjclark**
fixes #10357

### CODE REVIEW

File: lighthouse-core/lib/i18n/i18n.js [hunk 1]
```diff
@@ -56,8 +56,12 @@ const UIStrings = {
   displayValueMsSavings: 'Potential savings of {wastedMs, number, milliseconds}\xa0ms',
   /** Label for a column in a data table; entries will be the URL of a web resource */
   columnURL: 'URL',
-  /** Label for a column in a data table; entries will be the size of a web resource in kilobytes. */
+  /** Label for a column in a data table; entries will be the size of something resource (generic). */
```
Comment (connorjclark) [inline lighthouse-core/lib/i18n/i18n.js]: tap targets still uses this

File: lighthouse-core/lib/i18n/i18n.js [hunk 2]
```diff
@@ -72,8 +76,6 @@ const UIStrings = {
   columnResourceType: 'Resource Type',
   /** Label for a column in a data table; entries will be the number of network requests done by a webpage. */
   columnRequests: 'Requests',
-  /** Label for a column in a data table; entries will be the number of kilobytes transferred to load a set of files. */
-  columnTransferSize: 'Transfer Size',
```
Comment (connorjclark) [inline lighthouse-core/lib/i18n/i18n.js]: moved this to be with its friends. and tweaks the translator description a bit

**Comment by connorjclark** [comment]
fyi i want this in 6.0

**Comment by brendankenny** [comment]
> fyi i want this in 6.0

thanks for keeping us updated :P

File: lighthouse-core/lib/i18n/i18n.js [hunk 3]
```diff
@@ -56,9 +56,13 @@ const UIStrings = {
   displayValueMsSavings: 'Potential savings of {wastedMs, number, milliseconds}\xa0ms',
   /** Label for a column in a data table; entries will be the URL of a web resource */
   columnURL: 'URL',
-  /** Label for a column in a data table; entries will be the size of a web resource in kilobytes. */
+  /** Label for a column in a data table; entries will be the size of something resource (generic). */
```
Comment (brendankenny) [inline lighthouse-core/lib/i18n/i18n.js]:
```suggestion
  /** Label for a column in a data table; entries will be the size or quantity of some resource, e.g. the width and height dimensions of an image or the number of images in a web page. */
```
or some broad-enough set of examples like that.

File: lighthouse-core/lib/i18n/i18n.js [hunk 4]
```diff
@@ -56,9 +56,13 @@ const UIStrings = {
...
   columnSize: 'Size',
+  /** Label for a column in a data table; entries will be the transfer size (on the wire) of a web resource in kilobytes. */
```
Comment (brendankenny) [inline lighthouse-core/lib/i18n/i18n.js]: Looks like these descriptions got switched around?

File: lighthouse-core/lib/i18n/i18n.js [hunk 5]
```diff
@@ -56,9 +56,13 @@ const UIStrings = {
...
+  columnResourceSize: 'Resource Size',
```
Comment (brendankenny) [inline lighthouse-core/lib/i18n/i18n.js]: in https://github.com/GoogleChrome/lighthouse/issues/10357#issuecomment-591075172 it looks like the consensus might have been "File Size"? Even though we use resourceSize internally, I can see the argument for clearly communicating with end users

File: lighthouse-core/lib/i18n/i18n.js [hunk 6]
```diff
@@ -56,9 +56,13 @@ const UIStrings = {
...
   /** Label for a column in a data table; entries will be the time to live value of the cache header on a web resource. */
+  columnTransferSize: 'Transfer Size',
+  /** Label for a column in a data table; entries will be the resource size (post-decoding) of a web resource in kilobytes. */
```
Comment (brendankenny) [inline lighthouse-core/lib/i18n/i18n.js]: The previous description's "number of kilobytes transferred to load a set of files" (and something like "the size of a file in kilobytes") might be more meaningful to translators than "on the wire" and "post-decoding".

File: lighthouse-core/lib/i18n/i18n.js (hunk 5 above)
Comment (connorjclark) [reply lighthouse-core/lib/i18n/i18n.js]: In web parlance, is "file" really more clear than "resource"? Are files a thing, really?

File: lighthouse-core/lib/i18n/i18n.js (hunk 6 above)
Comment (connorjclark) [reply lighthouse-core/lib/i18n/i18n.js]: switched to "file size" and "download size" in these strings

File: lighthouse-core/lib/i18n/i18n.js (hunk 5 above)
Comment (connorjclark) [reply lighthouse-core/lib/i18n/i18n.js]: we decided resource size. also, btw CDT says resource size in network panel, so thats a strong reason to go with resource.

**Comment by brendankenny** [comment]
no idea why the travis status is pending even though it's green as of 24 days ago when you click through. Restarting...

**Pull request merged**